### ISSUE TITLE
core: incompatible constraint is displayed at the wrong place along the path

### ISSUE THREAD

**Issue by elise-chin** [OpenRailAssociation/osrd#9110]
### What happened?

Probable cause: There might be an important offset between the GPS coordinates of a track and its length reported in the infrastructure file: this probably leads to the following issue where a range of incompatible constraint is displayed at the wrong place on the path.
With @bougue-pe 

A gauge incompatible constraint is observed on the line from Tarascon to Sète-Ville.
![Capture d’écran 2024-09-27 à 16 54 19 (1)](https://github.com/user-attachments/assets/233c5812-feef-461a-ac06-a4ce18d16468)

Several investigations were done:
1. Simulation on the probable problematic portion
![Capture d’écran 2024-09-27 à 16 57 54](https://github.com/user-attachments/assets/6fc76688-6763-47fb-85aa-dd923597cea5)
There is no pathfinding error

2. Comparison between the track section and rolling stock gauges
     - Rolling stock 1TGVDUPL: FR3.3
     - Track section `1d3b6b18-0bbb-11ea-80ff-01f06fb51c27`: accepts FR3.3
     
=> It seems that the problematic track section isn't displayed correctly on the map. 

Now, if we zoom on the route, starting to Nîmes, it seems that we have found the real incompatible gauge constraint:
<img width="975" alt="Capture d’écran 2024-09-30 à 14 39 12" src="https://github.com/user-attachments/assets/56ccf3ea-0998-4779-a1dd-b62d811f9bb6">

The corresponding track sections (`61ea6d80-6667-11e3-81ff-01f464e0362d`, `61ea711a-6667-11e3-81ff-01f464e0362d`, `61ea7506-6667-11e3-81ff-01f464e0362d`) have G1 and GB gauge limits, which are indeed incompatible with FR3.3.

This offset can also be observed for Sète OP: the real GPS coordinates do not match the ones output by the pathfinding computation. 
![image (11)](https://github.com/user-attachments/assets/a850945d-c790-403c-ac04-a5d9fba48f92)

### What did you expect to happen?

Have the problematic section displayed where it actually is: ~200m downstream of the path than now.

### How can we reproduce it (as minimally and precisely as possible)?

- Infra: `Sharepoint / Données / Fichiers / 0-OSRD Dumps / Raw Infra / [infra for issue #9110] Brieuc_ddr_11.zip`
- Rolling stock: `1TGVDUPL`
- Route: `LYD VCV NI MPL BS NA CCE TE`

### On which environments the bug occurs?

Local

### On which browser the bug occurs?

Google Chrome / Firefox

### OSRD version (top right corner `Account` button > `Informations`)

a5e9ca26575d885868a4e4a947479fea37c7299e

**Comment by bougue-pe** [comment]
@maelysLeratRosso should look at this to help decide if it should be handled shortly or not.

Currently, we would advise the operational-study/infra-editor users to:
* shorten the path to "zoom" on the precise problematic section to make sure it is displayed with the minimum offset
* if there is an offset, to locate the problematic track-section where length doesn't match GPS, dichotomy on the start of the path should be done. That second part is not the responsibility of the pathfinding anyway I guess, and would probably lead to a separate US to add a warning on track sections where the length is "really far" from what the GPS coordinates suggest.

**Comment by SharglutDev** [comment]
@maelysLeratRosso ping :)

**Comment by bougue-pe** [comment]
For the record, a workshop is planned (the 6th of February for now).

The goals for this are :
* make sure this is an actual issue (can/will be handled elsewhere?)
* build a "generic" target for those cases (based on ROI)
* then only after that, determine if/where it should be implemented

**Comment by bougue-pe** [comment]
New plan after discussion in https://github.com/OpenRailAssociation/osrd/issues/9372#issuecomment-2765815821:
* work on this only after fixing #9228 (implem editoast part, and front converter based on editoast's response)
* change front: when currently using directly topo value for the bounds of intervals, first convert it before using it to display on the path shape.

---

Deprecated plan:
> After discussions in https://github.com/OpenRailAssociation/osrd/issues/9372#issuecomment-2639484994, the plan is to fix this by:
> * following the TLDR (add geographical ratio to the topological offsets/ranges in the response)
> * work on this only after fixing #9228 (it may help choose to move ratio computing implem in editoast if core is too difficult)
> * implem ratio computing where it's easier (probably core)
> * change front to use ratio

**Comment by SharglutDev** [comment]
Tried to reproduce the bug with a Tarascon - Sète and the same MR as the screen and didn't get an incompatible constraint error.

Can you still reproduce @bougue-pe ?

**Comment by bougue-pe** [comment]
Fully edited:

I think it's still bugged, I will try to reproduce (difficult to get a constraint error, but the bug should still be there).

**Comment by bougue-pe** [comment]
Confirmed that it's still there.

⚠️ the calculated length also seems wrong here : we should use topo length, not geographical one like it seems to be the case.

On small-infra destination is on the buffer stop of the northest lane of `WS`:
* Using just the very end (origin at the switch) with a `1TGVDUPL` we get this (closer to expected behavior):
  <img width="1872" height="840" alt="Image" src="https://github.com/user-attachments/assets/75abe884-9cba-4ea4-9806-b394718985d6" />
* Whereas if origin is at `NES` we get that:
  <img width="1862" height="837" alt="Image" src="https://github.com/user-attachments/assets/6be6661a-8358-47e2-ba66-6e516bcc097f" />


Also on prod for `France 2026-04-16 (Sennecey-le-Grand) (+) | ID48` :
* When starting from Nîmes using a `2TGVDUPL` and going to Lisle-sur-Tarn, we get this incompatible elec display (starting BEFORE `396+635`)
  <img width="1487" height="723" alt="Image" src="https://github.com/user-attachments/assets/f7bdc82c-2902-45b5-ba3a-fc837352afc0" />
* When starting later from Carcassonne we get this one (starting AFTER `396+635`)
  <img width="1478" height="710" alt="Image" src="https://github.com/user-attachments/assets/8f394400-7afb-4cda-a2e8-254fd484b6b3" />

On this infra, the main zone where topo and geo lengths diverge seem to be between `BS` and `NA`.
